src/metabase/driver/sqlserver.clj

```diff
@@ -0,0 +1,144 @@
+(ns metabase.driver.sqlserver
+  (:require [clojure.string :as s]
+            (korma [core :as k]
+                   [db :as kdb])
+            [korma.sql.utils :as utils]
+            [metabase.driver :refer [defdriver]]
+            [metabase.driver.generic-sql :refer [sql-driver]]
+            [metabase.driver.generic-sql.util :refer [funcs]])
+  (:import net.sourceforge.jtds.jdbc.Driver)) ; need to import this in order to load JDBC driver
+
+(def ^:private ^:const column->base-type
+  "See [this page](https://msdn.microsoft.com/en-us/library/ms187752.aspx) for details."
+  {:bigint           :BigIntegerField
+   :binary           :UnknownField
+   :bit              :BooleanField     ; actually this is 1 / 0 instead of true / false ...
+   :char             :CharField
+   :cursor           :UnknownField
+   :date             :DateField
+   :datetime         :DateTimeField
+   :datetime2        :DateTimeField
+   :datetimeoffset   :DateTimeField
+   :decimal          :DecimalField
+   :float            :FloatField
+   :geography        :UnknownField
+   :geometry         :UnknownField
+   :hierarchyid      :UnknownField
+   :image            :UnknownField
+   :int              :IntegerField
+   :money            :DecimalField
+   :nchar            :CharField
+   :ntext            :TextField
+   :numeric          :DecimalField
+   :nvarchar         :TextField
+   :real             :FloatField
+   :smalldatetime    :DateTimeField
+   :smallint         :IntegerField
+   :smallmoney       :DecimalField
+   :sql_variant      :UnknownField
+   :table            :UnknownField
+   :text             :TextField
+   :time             :TimeField
+   :timestamp        :UnknownField          ; not a standard SQL timestamp, see https://msdn.microsoft.com/en-us/library/ms182776.aspx
+   :tinyint          :IntegerField
+   :uniqueidentifier :UUIDField
+   :varbinary        :UnknownField
+   :varchar          :TextField
+   :xml              :UnknownField
+   (keyword "int identity") :IntegerField}) ; auto-incrementing integer (ie pk) field
+
+(defn- connection-details->spec [details]
+  (-> (kdb/mssql details)
+      ;; swap out Microsoft Driver details for jTDS ones
+      (assoc :classname   "net.sourceforge.jtds.jdbc.Driver"
+             :subprotocol "jtds:sqlserver")
+      ;; adjust the connection URL to match up with the jTDS format (see http://jtds.sourceforge.net/faq.html#urlFormat)
+      (update :subname #(s/replace % #";database=" "/"))))
```

Comment:
this seems like it could be dangerous because we are assuming that `;database=` is the first argument on the connection string.  is that something that's guaranteed?